Modern Dark Console

RedDevilus opened the pull request:
Adds a modern dark mode to the console that is atleast usable while not burning my eye sockets.

lightningterror commented on pcsx2/gui/ConsoleLogger.cpp:
```diff
@@ -514,6 +548,11 @@ ConsoleLogFrame::ConsoleLogFrame( MainEmuFrame *parent, const wxString& title, A
 	{
 		pMenuBar->Check(MenuId_ColorScheme_Light, true);
 	}
+	else //if ((0==m_conf.Theme.CmpNoCase("Default")) || (0==m_conf.Theme.CmpNoCase("ClassicDark")))
```
Need to change the above statement to else if (0==m_conf.Theme.CmpNoCase("Light")))

refractionpcsx2 commented on pcsx2/gui/ConsoleLogger.cpp:
```diff
@@ -548,7 +548,7 @@ ConsoleLogFrame::ConsoleLogFrame( MainEmuFrame *parent, const wxString& title, A
 		pMenuBar->Check(MenuId_ColorScheme_Light, true);
 	}
-	else //if ((0==m_conf.Theme.CmpNoCase("Default")) || (0==m_conf.Theme.CmpNoCase("ClassicDark")))
+	else //if ((0==m_conf.Theme.CmpNoCase("Default")) || (0==m_conf.Theme.CmpNoCase("Light")))
```
That's not fixing the issue

refractionpcsx2 commented on pcsx2/gui/ConsoleLogger.cpp:
```diff
@@ -425,7 +458,8 @@ ConsoleLogFrame::ConsoleLogFrame( MainEmuFrame *parent, const wxString& title, A
 	menuFontSizes.Append( MenuId_ColorScheme_Light,	_("&Light theme"),	_t("Default soft-tone color scheme."), wxITEM_RADIO );
-	menuFontSizes.Append( MenuId_ColorScheme_Dark,	_("&Dark theme"),	_t("Classic black color scheme for people who enjoy having text seared into their optic nerves."), wxITEM_RADIO );
+	menuFontSizes.Append( MenuId_ColorScheme_Dark,	_("&Dark theme"),	_t("Modern twist on Dark Theme which does not burn optic nerves."), wxITEM_RADIO );
+	menuFontSizes.Append( MenuId_ColorScheme_ClassicDark,	_("&Classic Dark theme"),	_t("Classic black color scheme for people who enjoy having text seared into their optic nerves."), wxITEM_RADIO );
```
Seemed to have deleted my comment accidentally.  Inappropriate descriptions. Please use something like "Modern dark theme" and "Classic PCSX2 style dark theme"

RedDevilus commented:
![image](https://user-images.githubusercontent.com/24227051/83960447-45028380-a889-11ea-8d82-b60a38c8b64f.png)
WX is breaking my commits, i break WX (by destroying old dark mode) and its fine again.
Needs some investigation

RedDevilus commented:
ConsoleLogger.cpp(513,112): message : Conversion from integral type to pointer type requires reinterpret_cast, C-style cast or function-style cast. When i add it to the bind events.

RedDevilus commented:
[ConsoleLogger.zip](https://github.com/PCSX2/pcsx2/files/4742874/ConsoleLogger.zip)
Original file. 

Also need to fix dark mode (replaced the colors)

RedDevilus commented:
Preview of other colors 
![image](https://user-images.githubusercontent.com/24227051/83979892-ef25ee00-a911-11ea-9bfc-f87a9bf96ccf.png)

RedDevilus commented:
Fuchsia for this line because it isn't an error
![image](https://user-images.githubusercontent.com/24227051/83979986-74110780-a912-11ea-94e8-9f22fdd92f8e.png)

MrCK1 commented:
I think the colors should match the other themes as closely as possible, with any modified values being for visibiltiy/contrast purposes.

RedDevilus commented:
Actually that line is from Red table so it does meet the same color

RedDevilus commented:
I changed it to only use 2 Themes (basically yeet the old dark mode but still have it commented)

RedDevilus commented:
![image](https://user-images.githubusercontent.com/24227051/83980815-f7cdf280-a918-11ea-9fc1-3b1b26d55886.png)
I upgraded the red to a pinkish one cuz Fuchsia isnt good

RedDevilus commented:
![image](https://user-images.githubusercontent.com/24227051/83980862-509d8b00-a919-11ea-98a4-4120e9341db2.png)

Perhaps i could make the blue a tad more light.

RedDevilus commented:
![image](https://user-images.githubusercontent.com/24227051/83980899-83e01a00-a919-11ea-9892-f68e00072457.png)
This is the Crimson Red color you get for errors. This seems alright to me.

RedDevilus commented:
![image](https://user-images.githubusercontent.com/24227051/84040025-5ee3b980-a9a2-11ea-8877-b47ff6b6455c.png)
I changed the name in the context(dropdown?) menu to make it obvious its a different theme. The old dark theme is commented out for a future PR to bring it back. Light theme is still the default look
Modern dark theme (new)
![image](https://user-images.githubusercontent.com/24227051/84040696-36a88a80-a9a3-11ea-9478-7bb5ac4f9390.png)

Default (light theme)
![image](https://user-images.githubusercontent.com/24227051/84040268-b2560780-a9a2-11ea-8fde-bd9d7d6385db.png)
This aged well considering its look.

Dark theme (deactivated/commented out)
![image](https://user-images.githubusercontent.com/24227051/84040429-e9c4b400-a9a2-11ea-94f9-4e706812229f.png)
Dark blue on black is one of the worst things you can do to the eyes, combine that with dark green and esspecialy bright yellow and your eyes will burn.

You do have to completely restart PCSX2 if you change the theme.

RedDevilus commented:
Several situations possible for this
Have 3 themes with light theme as default
Have 3 themes with dark theme as default
Have 2 themes with light theme as default (current commit)
Have 2 themes with dark theme as default

Unless someone wants to keep the option to have it in the GUI (it's commented out in the files itself)
Opinions, suggestions?

Other than that it's finished if you want to be it like the current commit.

RedDevilus commented:
After some discussion; the ideal situation would be 
Have 3 themes with light theme as default or have 3 themes with dark theme as default.
My current commit just replaced the RGB colors of the classic dark mode and moved the real classic dark mode into comments. 

I then added the others comments which i assumed controlled the dark theme through WX. This didn't work out and bear with me i have 0 WX knowledge and i don't really code. I just glanced at it and added those other commented out to follow my though progress what i tried in previous attempt.

lightningterror commented:
Personally I'd just replace the old dark theme with the new one, nobody will use the old one anyway.

lightningterror commented:
Status?

RedDevilus commented:
Have to look into how to compile debug PCSX2 correctly instead of cmake weirdness.

lightningterror commented:
Just overwrite old theme, no need for extra complexity.

RedDevilus commented:
![image](https://user-images.githubusercontent.com/24227051/87087887-2dc20780-c234-11ea-978a-24717f128163.png)
Current commit is what i said above, with appveyor artifact below.
[pcsx2-v1.7.0-dev-72-g55326d634-7706-vs2019-Win32-AppVeyor.zip](https://github.com/PCSX2/pcsx2/files/4899379/pcsx2-v1.7.0-dev-72-g55326d634-7706-vs2019-Win32-AppVeyor.zip)
out

RedDevilus commented:
I'll remove the commented out Classic dark mode, in effect replacing the code with the modern colors to reduce extra text for no reason. If someone needs it they can get it from 1.6.0, here or other locations.

lightningterror commented on pcsx2/gui/ConsoleLogger.cpp:
```diff
@@ -425,7 +425,7 @@ ConsoleLogFrame::ConsoleLogFrame( MainEmuFrame *parent, const wxString& title, A
 	menuFontSizes.AppendSeparator();
 	menuFontSizes.Append( MenuId_ColorScheme_Light,	_("&Light theme"),	_t("Default soft-tone color scheme."), wxITEM_RADIO );
-	menuFontSizes.Append( MenuId_ColorScheme_Dark,	_("&Dark theme"),	_t("Classic black color scheme for people who enjoy having text seared into their optic nerves."), wxITEM_RADIO );
+	menuFontSizes.Append( MenuId_ColorScheme_Dark,	_("&Modern Dark theme"),	_t("Modern twist on Dark Theme which does not burn optic nerves."), wxITEM_RADIO );
```
`Modern dark color scheme.` for tooltip sounds better and fits along with light theme one.

RedDevilus commented:
To summarise this PR will do:
Have 2 themes with light theme as default (current commit) and replacing the classic dark mode with my colors which are inspired on Discord colors. 

Black background is replaced with grey, warnings are crimson red and minor 'reminder warnings' like no controller connected with LilyPad (and 3.x GL context for some reason) are softer red. Yellow is a tad more yellow than brown on light theme. You can see the screenshots above anyway.

turtleli commented on pcsx2/gui/ConsoleLogger.cpp:
```diff
@@ -425,7 +425,7 @@ ConsoleLogFrame::ConsoleLogFrame( MainEmuFrame *parent, const wxString& title, A
 	menuFontSizes.AppendSeparator();
 	menuFontSizes.Append( MenuId_ColorScheme_Light,	_("&Light theme"),	_t("Default soft-tone color scheme."), wxITEM_RADIO );
-	menuFontSizes.Append( MenuId_ColorScheme_Dark,	_("&Dark theme"),	_t("Classic black color scheme for people who enjoy having text seared into their optic nerves."), wxITEM_RADIO );
+	menuFontSizes.Append( MenuId_ColorScheme_Dark,	_("&Modern Dark theme"),	_t("Modern dark color scheme which does not burn optic nerves."), wxITEM_RADIO );
```
I don't think you need to change the menu entry text (so just `Dark theme` is probably fine).
The menu help text could be reduced to just `Modern dark color scheme.` (which someone suggested earlier I think) - It's shorter and IMO it's better to avoid humour in translatable strings since it might not necessarily translate well.

RedDevilus replied:
Fair enough it was based on the Classic dark theme. I could do Dark theme (modern) for it's entry too.

RedDevilus commented:
![image](https://user-images.githubusercontent.com/24227051/87202102-2e27d480-c300-11ea-98c4-5a5e432f4784.png)
This is the end result.
